tests/flytekit/unit/core/test_type_engine.py

```diff
@@ -967,6 +967,7 @@ class TestFileStruct(DataClassJsonMixin):
         b: typing.Optional[FlyteFile]
         b_prime: typing.Optional[FlyteFile]
         c: typing.Union[FlyteFile, None]
+        c_prime: typing.Union[None, FlyteFile]
```

Comment:
Can you make this
```suggestion
        c_prime: typing.Union[None, StructuredDataset, int, FlyteFile]
```

Reply:
actually can you write one more unit test for me please?  (and add it under `test_dataclass.py` this file is getting too big).

```python
@dataclass
class A():
  x: str

@dataclass
class B():
   x: FlyteFile

```
then call `_make_dataclass_serializable` on `Union[A, B]` where `b = B(x="s3://tmp)` or something.